src/tools/kata-ctl/src/check.rs

```diff
@@ -5,8 +5,105 @@
 
 // Contains checks that are not architecture-specific
 
+//use anyhow::{anyhow, Error, Result};
 use anyhow::{Result};
+use std::fs;
+use std::error::Error;
+//use std::result::Result::Err;
+//use string_error::static_err;
+//use std::str;
+
+//struct CpuDetails {
+//    cpu_info_file: &str,
+//    required_cpu_flags:
+//    required_cpu_attribs:
+//}
+
+fn get_cpu_info(cpu_info_file: &str) -> Result<String> {
+    let contents = match fs::read_to_string(cpu_info_file) {
+        Ok(c) => c,
+        Err(e) => return Err(e)?,
+    };
+    Ok(contents)
+}
+
+// get_single_cpu_info returns the contents of the first cpu from
+// the specified cpuinfo file by parsing based on a specified delimiter
+pub fn get_single_cpu_info(cpu_info_file: &str, substring: &str) -> Result<String, Box<dyn Error>> {
+    let contents = match get_cpu_info(cpu_info_file) {
+        Ok(c) => c,
+        Err(e) => return Err(e)?,
+    };
+
+    if contents.is_empty() {
+        return Err("cpu_info file is empty")?;
+    }
+
+    let subcontents: Vec<&str> = contents.split(substring).collect();
+    let result = subcontents.first().unwrap().to_string();
+
+    Ok(result)
+}
+
+// get_cpu_flags returns a string of cpu flags from cpuinfo, passed in
+// as a string
+pub fn get_cpu_flags(cpu_info: &str, cpu_flags_tag: &str) -> Result<String, Box<dyn Error>> {
+    if cpu_info.is_empty() {
+        return Err("cpu_info string is empty")?;
+    }
+
+    let subcontents: Vec<&str> = cpu_info.split("\n").collect();
+    for line in subcontents {
+        if line.starts_with(cpu_flags_tag) {
+            let line_data: Vec<&str> = line.split(":").collect();
+            let flags = line_data.last().unwrap().to_string();
+            return Ok(flags);
+        }
+    }
+
+    Ok("".to_string())
+}
+
+pub fn check_cpu_flags(retrieved_flags: &str, required_flags: &str) -> Result<Vec<String>> {
```

Comment:
Add description: check_cpu_flags tests if required_flags are in the retrieved_flags and adds missing flags to a Vec to return for handling